src/Encoder/BlockCipher.js

```diff
@@ -0,0 +1,328 @@
+      {
+        name: 'algorithm',
+        type: 'enum',
+        value: 'aes-128',
+        randomizable: false,
+        options: {
+          elements: algorithms.map(algorithm => algorithm.name),
+          labels: algorithms.map(algorithm => algorithm.label)
+        }
+      },
+      {
+        name: 'mode',
+        type: 'enum',
+        value: 'cbc',
+        randomizable: false,
+        options: {
+          elements: modes.map(algorithm => algorithm.name),
+          labels: modes.map(algorithm => algorithm.label)
+        }
+      },
+      {
+        name: 'key',
+        type: 'bytes',
+        value: new Uint8Array([
+          0x2b, 0x7e, 0x15, 0x16, 0x28, 0xae, 0xd2, 0xa6,
+          0xab, 0xf7, 0x15, 0x88, 0x09, 0xcf, 0x4f, 0x3c ]
+        )
+      },
+      {
+        name: 'iv',
+        type: 'bytes',
+        value: new Uint8Array([
+          0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+          0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F ]
+        )
+      } /*,
+      {
+        name: 'pad',
+        type: 'boolean',
+        value: false
+      } */
+
+    ])
+  }
+
+  /**
+   * Triggered when a setting value has changed.
+   * Override is required to call super.
+   * @protected
+   * @param {Setting} setting
+   * @param {mixed} value Setting value
+   */
+  settingValueDidChange (setting, value) {
+    super.settingValueDidChange(setting, value)
+
+    if (setting.getName() === 'mode') {
+      let ivSetting = this.getSetting('iv')
+      const isVisible = (value !== 'ecb')
+      ivSetting && ivSetting.setVisible(isVisible)
+    }
+  }
+
+  /**
+   * Performs encode or decode on given content.
+   * @protected
+   * @param {Chain} content
+   * @param {boolean} isEncode True for encoding, false for decoding
+   * @return {Chain|Promise} Resulting content
+   */
+  async performTranslate (content, isEncode) {
+    const algorithmName = this.getSettingValue('algorithm')
+    const modeName = this.getSettingValue('mode')
+
+    if (!Browser.isNode() && (modeName === 'ecb')) {
+      throw new InvalidInputError(
+        `Algorithm ${algorithmName}-${modeName} not supported by WebCrypto`)
+    }
+
+    const keyLength = BlockCipherEncoder.getAlgorithmKeySize(algorithmName) / 8
+    const blockLength = BlockCipherEncoder.getAlgorithmBlockSize(algorithmName) / 8
+
+    const key = this.getSettingValue('key')
+    if (key.length !== keyLength) {
+      throw new InvalidInputError(
+        `Algorithm ${algorithmName} requires a key of '${keyLength}' bytes`)
+    }
+
+    let iv = this.getSettingValue('iv')
+    if (BlockCipherEncoder.getModeNeedsIV(modeName)) {
+      if (iv.length !== blockLength) {
+        throw new InvalidInputError(
+          `Algorithm ${algorithmName}-${modeName} requires an IV of '${blockLength}' bytes`)
+      }
+    } else {
+      iv = undefined
+    }
+
+    const cipherText = await this.createCipher(algorithmName, modeName, key, iv, false, isEncode, content.getBytes())
```

Comment:
`createCipher` will fail when sending invalid content to it (e.g. when decoding with a different algorithm or mode). You need to catch such an exception and put it into an `InvalidInputError` with a human readable message (which can be derived from the original one). This will make the message visible at the bottom of the brick while stopping it from being reported as implementation error to sentry.io.

Reply:
@ffraenz thanks for the feedback. Adjustments implemented as per your suggestions.

Reply:
@ffraenz - just one thing. AES-192 keys are not widely supported in Browsers, maybe this should be disabled for now. Unfortunately, for full support, we'll have to pull in an external JS library. Do you have any preference,  crypto-js or forge (https://github.com/digitalbazaar/forge) ?